src/sql/azdata.proposed.d.ts

```diff
@@ -29,6 +29,12 @@ declare module 'azdata' {
 		export function getConnection(uri: string): Thenable<ConnectionProfile>;
 	}
 
+	export interface NotebookDocument {
```

Comment:
This needs to be in the nb namespace as well (that's why you were getting the error about setTrusted not existing on the type). Once you do that you should be able to get rid of the warning suppression you put there